File: Beecrowd/1089/1089.py
while True:
    N = int(input())
    if N == 0:
        break
    wave = list(map(int, input().split()))
    wave.append(wave[0])

    peaks = 0
    for i in range(0, N):
        if wave[i] > wave[i-1] and wave[i] > wave[i+1]:
            peaks += 1
        if wave[i] < wave[i-1] and wave[i] < wave[i+1]:
            peaks += 1

    if wave[0] > wave[N-1] and wave[0] > wave[1]:
        peaks += 1
    if wave[0] < wave[N-1] and wave[0] < wave[1]:
        peaks += 1

    print(peaks)

File: Beecrowd/1089/1089.cpp
/* Problem statement: https://www.beecrowd.com.br/judge/en/problems/view/1089 */
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int N, H, peaks;
    vector<int>wave;
    while(cin >> N && N != 0){
        for(int i = 0; i < N; i++){
            cin >> H;
            wave.push_back(H);
        }
        wave.push_back(wave[0]);
        peaks = 0;
        for(int j = 1; j < N; j++){
            if(wave[j] > wave[j-1] && wave[j] > wave[j+1]){
                peaks++;
            }
            if(wave[j] < wave[j-1] && wave[j] < wave[j+1]){
                peaks++;
            }
        }
        if(wave[0] > wave[N-1] && wave[0] > wave[1]){ peaks++; }
        if(wave[0] < wave[N-1] && wave[0] < wave[1]){ peaks++; }
        
        cout << peaks << endl;
        wave.clear();
    }

    return 0;
}

